tests/Doctrine/Tests/Common/Collections/ClosureExpressionVisitorTest.php

```diff
@@ -190,6 +190,17 @@ public function testWalkEndsWithComparison() : void
         self::assertFalse($closure(new TestObject('hello')));
     }
 
+    public function testWalkBitAndComparison() : void
+    {
+        $closure = $this->visitor->walkComparison($this->builder->bitAnd('foo', 4));
+
+        self::assertTrue($closure(new TestObject(4)));
+        self::assertTrue($closure(new TestObject('4')));
+        self::assertTrue($closure(new TestObject(5)));
+        self::assertTrue($closure(new TestObject('5')));
+        self::assertFalse($closure(new TestObject(2)));
```

Comment:
Please add a few more false cases like `8` or `11`. Be sure that there are also false-cases with a string-number like in your true-cases.